src/components/TodoApp.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const TodoApp = ({ setTodoList }) => {
+  const [todo, setTodo] = useState('');
+
+  const addTodo = () => {
+    if (!todo) {
+      return;
+    }
+
+    setTodoList(prevTodos => (
+      [
+        ...prevTodos,
+        {
+          id: +new Date(),
+          title: todo.trimRight(),
+          completed: false,
+        },
+      ]
+    ));
+    setTodo('');
+  };
+
+  return (
+    <form
+      onSubmit={(event) => {
+        event.preventDefault();
+        addTodo();
+      }}
+    >
+      <input
+        type="text"
+        className="new-todo"
+        placeholder="What needs to be done?"
+        value={todo}
+        onChange={event => setTodo(event.target.value.trimLeft())}
```

Comment:
trim for both side

Reply:
not fixed

Reply:
Fixed but in later update (Editing added). But now I understand that this is not necessary, because if you want to add for example "Learn(space)JS" you can't do it for obvious reasons